src/ui/features/onboarding/initialization.ts

```diff
@@ -0,0 +1,34 @@
+import { getCurrentUser } from 'src/shared/getCurrentUser';
+import { openOnboarding } from 'src/shared/openOnboarding';
+import { UrlContextParam, urlContext } from 'src/ui/shared/UrlContext';
+import { setUrlContext } from 'src/ui/shared/setUrlContext';
+import { OnboardingInterrupt } from './errors';
+
+async function getAppMode() {
+  const hasOnboardingUrl = document.location.hash.startsWith('#/onboarding');
+  const hasExistingUser = Boolean(await getCurrentUser());
```

Comment:
You're querying `await getCurrentUser()` twice. Let's pass `{ hasExistingUser: boolean }` as a param instead? This will also make this function synchronous.

Reply:
yep, thats better, thanks ✅